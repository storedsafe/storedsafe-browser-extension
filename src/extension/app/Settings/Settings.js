import React from 'react';

export default function Settings(props) {
  const fields = props.fields.map(field => {
    return (
      <label key={field.key} htmlFor={field.key}>
        <p>{field.title}:</p>
        {React.createElement(field.element, {
          ...field.attr,
          [field.event]: props.onValueChange,
          id: field.key,
          name: field.key,
          disabled: field.managed === true,
        })}
      </label>
    )
  });

  return (
    <section className="settings">
      {fields}
    </section>
  );
}
