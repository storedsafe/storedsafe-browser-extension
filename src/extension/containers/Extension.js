import React from 'react';

export default function Extension(props) {
  const path = window.location.href.split('#')[1];
  const breadcrumbs = path.split('/');

  const breadcrumbsLi = breadcrumbs.map(crumb => {
    return <li key={crumb.toLowerCase()}>{crumb}</li>;
  });

  return (
    <main>
      <p>Path</p>
      <ul>{breadcrumbsLi}</ul>
    </main>
  );
}
