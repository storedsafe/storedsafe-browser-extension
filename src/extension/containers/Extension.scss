@use '../base.scss';

* {
  padding: 0;
  margin: 0;
}

body {
  background-color: base.$bg-dark;
  color: base.$fg-light;
}
