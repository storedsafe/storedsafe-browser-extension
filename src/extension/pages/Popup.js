import React, { useState, useEffect } from 'react';

export default function Popup(props) {
  const [token, setToken] = useState(null);
  const [fields, setFields] = useState(null);

  const onSubmit = (event) => {
    event.preventDefault();
    setToken(event.target.token.value);
  };

  const handleMessage = (request, sender) => {
    setFields(Object.keys(request.fields).map(key => {
      if (/pass/.test(key)) {
        return <li key={key}>{request.fields[key].replace(/./g,'*')}</li>;
      }
      return <li key={key}>{request.fields[key]}</li>;
    }));
  };

  useEffect(() => {
    browser.runtime.onMessage.addListener(handleMessage);
  }, []);

  useEffect(() => {
    browser.storage.local.get('token')
      .then(values => {
        if(values.token !== undefined) {
          setToken(values.token.value);
        }
      });
  }, []);

  useEffect(() => {
    browser.storage.local.set({
      token: {
        value: token,
        lastRefresh: Date.now(),
      },
    });
  }, [token]);

  return (
    <section>
      <p>Token: {token}</p>
      <form onSubmit={onSubmit}>
        <input type="text" name="token" id="token" />
        <input type="submit" />
      </form>
      <ul>
        {fields}
      </ul>
    </section>
  );
}
