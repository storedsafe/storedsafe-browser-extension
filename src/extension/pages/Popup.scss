@use '../base.scss';

.popup {
  box-sizing: border-box;
  overflow: auto;
  width: 400px;
}

