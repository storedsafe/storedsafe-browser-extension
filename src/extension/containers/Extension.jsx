import React from 'react';
import Welcome from '../pages/Welcome';
import Popup from '../pages/Popup';
import './Extension.scss';

export default function Extension() {
  let page;
  const path = window.location.href.split('#')[1];

  switch (path) {
    case 'popup':
      page = <Popup />;
      break;
    default:
      page = <Welcome />;
  }

  return (
    <main className="extension">
      {page}
    </main>
  );
}
