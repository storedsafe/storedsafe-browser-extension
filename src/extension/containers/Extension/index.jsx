import React from 'react';
import Settings from '../../pages/Settings';
import Popup from '../../pages/Popup';

export default function Extension() {
  let page;
  const path = window.location.href.split('#')[1];

  switch (path) {
    case 'popup':
      page = <Popup />;
      break;
    default:
      page = <Settings />;
  }

  return (
    <main className="extension">
      {page}
    </main>
  );
}
