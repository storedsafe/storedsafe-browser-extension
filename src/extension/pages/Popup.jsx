import React, { useState, useEffect } from 'react';
import AuthForm from '../components/AuthForm';
import FilterableVaultList from '../components/FilterableVaultList';
import SaveLoginForm from '../components/SaveLoginForm';
import logo from '../../assets/logo.png';
import './Popup.scss';

export default function Popup(props) {
  return (
    <section className="popup">
      <header><img src={logo} alt="StoredSafe" /></header>
      <FilterableVaultList />
      <SaveLoginForm />
      <AuthForm />
    </section>
  );
}
