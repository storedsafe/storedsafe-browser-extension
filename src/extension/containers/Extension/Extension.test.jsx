import React from 'react';
import { shallow } from 'enzyme';
import Extension from '.';
import Settings from '../../pages/Settings';
import Popup from '../../pages/Popup';

global.window = Object.create(window);
Object.defineProperty(window, 'location', { value: { href: null } });

const url = 'extension://abc123/index.html';

test('<Extension />, default => <Settings />', () => {
  window.location.href = url;
  const wrapper = shallow(<Extension />);
  expect(wrapper.find(Settings).length).toBe(1);
  expect(wrapper).toMatchSnapshot();
});

test('<Extension />, #popup => <Popup />', () => {
  window.location.href = `${url}#popup`;
  const wrapper = shallow(<Extension />);
  expect(wrapper.find(Popup).length).toBe(1);
  expect(wrapper).toMatchSnapshot();
});
