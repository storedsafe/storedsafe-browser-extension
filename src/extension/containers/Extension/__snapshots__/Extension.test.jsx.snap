// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Extension />, #popup => <Popup /> 1`] = `ShallowWrapper {}`;

exports[`<Extension />, default => <Settings /> 1`] = `ShallowWrapper {}`;
